Pi2BoardPkg/Drivers/DisplayDxe/DisplayDxe.c

```diff
@@ -202,6 +202,81 @@ DisplaySetMode(
     return EFI_SUCCESS;
 }
 
+BOOLEAN TruncateBltRange(
+    IN UINT32 GopUpperBound,
+    IN UINTN BltOffset,
+    IN OUT UINTN *BltRange,
+    OUT EFI_STATUS *Status
+)
+{
+    *Status = EFI_SUCCESS;
+    if (BltOffset + *BltRange > GopUpperBound) {
+        if (GopUpperBound > BltOffset) {
+            *BltRange = GopUpperBound - BltOffset;
+            return TRUE;
+        } else {
+            *Status = EFI_INVALID_PARAMETER;
```

Comment:
What is the reason for returning a status result in this function?  If the ranges are not checked elsewhere for validity, then it should be added to DisplayBlt(), not here.

Reply:
I'm just trying to validation code separate from DisplayBlt(). DisplayBlt() is already quite large.

Reply:
Validation should be done as close to the upper API surface as possible.  Not nested 6 layers deep.  You've added new validation and now expanded the size of DisplayBlt to handle the extra status code returned from your sub functions.